bundles/framework/announcements/instance.js

```diff
@@ -0,0 +1,216 @@
+/**
+ * @class Oskari.framework.bundle.announcements.AnnouncementsBundleInstance
+ *
+ * Main component and starting point for the announcements functionality.
+ *
+ * See Oskari.framework.bundle.announcements.AnnouncementsBundleInstance for bundle definition.
+ */
+Oskari.clazz.define('Oskari.framework.bundle.announcements.AnnouncementsBundleInstance',
+
+    /**
+     * @method create called automatically on construction
+     * @static
+     */
+    function () {
+        this.sandbox = null;
+        this.started = false;
+        this.plugins = {};
+        this.localization = null;
+        this.seutumaisaSearchService = null;
+    }, {
+        /**
+         * @static
+         * @property __name
+         */
+        __name: 'announcements',
+        /**
+         * @method getName
+         * @return {String} the name for the component
+         */
+        getName: function () {
+            return this.__name;
+        },
+        /**
+         * @method setSandbox
+         * @param {Oskari.Sandbox} sandbox
+         * Sets the sandbox reference to this component
+         */
+        setSandbox: function (sandbox) {
+            this.sandbox = sandbox;
+        },
+        /**
+         * @method getSandbox
+         * @return {Oskari.Sandbox}
+         */
+        getSandbox: function () {
+            return this.sandbox;
+        },
+
+        /**
+         * @method getLocalization
+         * Returns JSON presentation of bundles localization data for current language.
+         * If key-parameter is not given, returns the whole localization data.
+         *
+         * @param {String} key (optional) if given, returns the value for key
+         * @return {String/Object} returns single localization string or
+         *     JSON object for complete data depending on localization
+         *     structure and if parameter key is given
+         */
+        getLocalization: function (key) {
+            if (!this._localization) {
+                this._localization = Oskari.getLocalization(this.getName());
+            }
+            if (key) {
+                return this._localization[key];
+            }
+            return this._localization;
+        },
+
+        /**
+         * @method start
+         * implements BundleInstance protocol start method
+         */
+        start: function () {
+            const conf = this.conf;
+            const sandboxName = conf ? conf.sandbox : 'sandbox';
+            const sandbox = Oskari.getSandbox(sandboxName);
+
+            if (this.started) {
+                return;
+            }
+
+            this.started = true;
+            this.sandbox = sandbox;
+
+            var announcementService = Oskari.clazz.create('Oskari.framework.bundle.announcements.AnnouncementsService', sandbox, this.getLocalization().service);
+            this.sandbox.registerService(announcementService);
```

Comment:
So both the admin-announcements and the announcement bundle create an instance of the service and register it to sandbox. This creates a conflict in the code that probably causes bugs at some point if not now. You should probably check if the service is already registered in sandbox before creating and registering a new instance.

Reply:
Looking at the service I'm not sure if it needs to be registered on sandbox at all as it's just a wrapper for XHR-calls.

Reply:
Service no longer created/used